sdcflows/interfaces/fmap.py

```diff
@@ -23,6 +24,111 @@
 
 LOGGER = logging.getLogger('nipype.interface')
 
+class ProcessPhasesInputSpec(BaseInterfaceInputSpec):
+    phase1_file = File(exists=True, mandatory=True, desc='phase1 file')
+    phase2_file = File(exists=True, mandatory=True, desc='phase2 file')
+    phase1_metadata = traits.Dict(mandatory=True, desc='phase1 metadata dict')
+    phase2_metadata = traits.Dict(mandatory=True, desc='phase2 metadata dict')
+
+
+class ProcessPhasesOutputSpec(TraitedSpec):
+    short_te_phase_image = File(exists=True, desc='short TE phase image scaled for unwrapping')
+    short_te_phase_metadata = traits.Dict(desc='short TE phase image metadata')
+    long_te_phase_image = File(exists=True, desc='long TE phase image scaled for unwrapping')
+    long_te_phase_metadata = traits.Dict(desc='long TE phase image metadata')
+    phasediff_metadata = traits.Dict(desc='the phasediff metadata')
+
+
+class ProcessPhases(SimpleInterface):
```

Comment:
What about splitting the two logical steps in this one? (i.e., one interface for the rescaling that can go in a MapNode and another one to reorder echoes&images and calculate the echo time diff)

Reply:
That makes sense. However, we'd need to keep track of the accompanying metadata. Maybe in a future PR?